corehq/motech/repeaters/models.py

```diff
@@ -1292,68 +1291,6 @@ def get_payload(repeater: Repeater, repeat_record: RepeatRecord) -> str:
         raise
 
 
-def send_request(
-    repeater: Repeater,
-    repeat_record: RepeatRecord,
-    payload: Any,
-) -> bool:
-    """
-    Calls ``repeater.send_request()`` and handles the result.
-
-    Returns True on success or cancelled, which means the caller should
-    not retry. False means a retry should be attempted later.
-    """
-
-    def is_success(resp):
-        return (
-            is_response(resp)
-            and 200 <= resp.status_code < 300
-            # `response` is `True` if the payload did not need to be
-            # sent. (This can happen, for example, with DHIS2 if the
-            # form that triggered the forwarder doesn't contain data
-            # for a DHIS2 Event.)
-            or resp is True
-        )
-
-    def allow_retries(response):
-        # respect the `retry` field of RepeaterResponse
-        return getattr(response, 'retry', True)
-
-    def later_might_be_better(resp):
-        return is_response(resp) and resp.status_code in (
-            502,  # Bad Gateway
-            503,  # Service Unavailable
-            504,  # Gateway Timeout
-        )
```

Comment:
@kaapstorm given we were never using this function in production code (just tests), am I correct in saying that these improvements aren't part of the current repeater flow? In that [any unsuccessful request](https://github.com/dimagi/commcare-hq/blob/178ef4575ceefa3e4da54fba3a940f976b0105ce/corehq/motech/repeaters/models.py#L483-L486) that doesn't result in an exception being raised is treated as a [server error,](https://github.com/dimagi/commcare-hq/blob/178ef4575ceefa3e4da54fba3a940f976b0105ce/corehq/motech/repeaters/models.py#L1208-L1209) meaning even 4xx related errors will get the max number of retries?